chore(broker): renamed payload to variables

MiguelPires opened the pull request:
Not everything was renamed because there are still some meaningful changes that need to be done, like getting rid of the payload column family in the VariablesState (#2225). After that we can either have an issue to mop up remaining occurrences of the old naming/concept or just apply the boy scout rule and rename them whenever we stumble upon them.  

closes #2194, #2195, #2215

Zelldon commented on clients/go/commands/publishMessage_command.go:
```diff
@@ -16,121 +16,121 @@
-	"time"
+    "context"
+    "fmt"
+    "github.com/zeebe-io/zeebe/clients/go/pb"
```
@menski do we have to somehow specify a rule for formatting like whitespaces instead of tabs  or idk ? :man_shrugging:

menski replied:
running `gofmt` generates the expected format

Zelldon commented on gossip/src/main/resources/gossip-schema.xml:
```diff
@@ -57,7 +57,7 @@
       <field name="senderId" id="12" type="uint16"/>
       <data name="eventType" id="14" type="varDataEncoding"/>
-      <data name="payload" id="15" type="varDataEncoding"/>
+      <data name="variables" id="15" type="varDataEncoding"/>
```
Ok here it is not necessary :sweat_smile: but since we remove this anyway just leave it as it is.

menski commented:
Please make sure that we mark these commits as breaking change, as they impact users.

Zelldon commented:
Yes that is just what I was going to say thanks :+1:

MiguelPires commented:
Thanks @Zelldon. I didn't rename everything because it seems like the old concept is still being used and I thought that it might be better to rename once we're sure everything can be renamed. Also, I did search for usages of payload and but some actually make sense to me. For instance, one of the files you mentioned (JobWorkerBuilderStep1) has something like "(...) only contain variables from this list in their payload." which I would argue is actually correct. I can reword it to not use "payload" but it doesn't sound wrong to me as it is. In any case, I'll change all of it besides VariablesState.

MiguelPires commented:
Hey @Zelldon, I addressed your comments. The things that weren't changed are:
* In the gossip module (since it will be deleted)
* Stuff related to raft/transport where payload isn't a payload in the "Zeebe" sense and more like a message payload
* Stuff related to the VariablesState payload/mapping thing (which according to https://github.com/zeebe-io/zeebe/issues/2225#issuecomment-475579059 doesn't need to be changed)

Zelldon commented on broker-core/src/main/java/io/zeebe/broker/workflow/state/EventScopeInstanceState.java:
```diff
@@ -170,7 +170,7 @@ private void createTrigger(
    * the event trigger from the state.
    *
-   * @param eventScopeKey the key of the event scope
+   * @param eventScopeKey the key of the event scopen
```
I assume this was an mistake :)

Zelldon replied:
```suggestion
   * @param eventScopeKey the key of the event scope
```

MiguelPires commented on gateway/pom.xml:
```diff
@@ -53,11 +55,6 @@
 
-    <dependency>
-      <groupId>io.zeebe</groupId>
-      <artifactId>zb-msgpack-core</artifactId>
```
No, I deleted the jmh folder at Menski's request so that dependency is no longer needed

Zelldon replied:
ok thanks

Zelldon commented:
I rebased the PR to the latest develop. Please squash the commits and write the BREAKING CHANGE message in the commit message. After that I will accept the PR

MiguelPires commented:
Ok, thanks. I squashed and added the breaking change flag. Please let me know if it should be more descriptive or if it has any issues
